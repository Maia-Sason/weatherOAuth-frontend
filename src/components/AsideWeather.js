const AsideWeather = () => {
  return <> </>;
};

export default AsideWeather;
