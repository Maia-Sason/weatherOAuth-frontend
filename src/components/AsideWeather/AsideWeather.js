import { Paper, Typography, Divider, makeStyles, Box } from "@material-ui/core";
import { ReactComponent as DropPercentage } from "../../icons/drop-percentage.svg";
import { ReactComponent as Wind } from "../../icons/wind-2.svg";

const useStyles = makeStyles((theme) => ({
  container: {
    background: "none",
    padding: "10px",
    //minHeight: "40vh",
    margin: "20px",
  },
  headerBox: {
    display: "flex",
    justifyContent: "space-between",
  },
  header: {
    fontWeight: 200,
    fontSize: "2em",
    color: "white",
  },
  headerB: {
    fontWeight: 200,
    fontSize: "3em",
    color: "white",
  },
  rainDrop: {
    height: "3em",
    width: "3em",
    fill: "white",
    margin: "10px",
  },
  precipitation: {
    padding: "10px",
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    //height: "100%",
  },
  wind: {
    padding: "10px",
    display: "flex",
    justifyContent: "space-between",
    alignItems: "center",
    //height: "100%",
  },
}));

const AsideWeather = ({ feels, rain, wind }) => {
  const classes = useStyles();

  return (
    <>
      <Paper elevation={10} className={classes.container}>
        <Box className={classes.headerBox}>
          <Typography className={classes.header}>Feels Like:</Typography>
          <Typography className={classes.header}>{feels}&deg;</Typography>
        </Box>
        <Divider />
        <Box className={classes.precipitation}>
          <DropPercentage className={classes.rainDrop} />
          <Typography className={classes.headerB}>{rain * 100}%</Typography>
        </Box>
        <Divider />
        <Box className={classes.wind}>
          <Wind className={classes.rainDrop} />
          <Typography className={classes.headerB}>{wind}mph</Typography>
        </Box>
      </Paper>
    </>
  );
};

export default AsideWeather;
