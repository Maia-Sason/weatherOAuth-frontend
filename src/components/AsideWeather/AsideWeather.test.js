import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import AsideWeather from "./AsideWeather";

test("<AsideWeather/> matches snapshot", () => {
  const component = render(<AsideWeather />);
  expect(component.container).toMatchSnapshot();
});

it("Renders AsideWeather without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<AsideWeather />, div);
});
