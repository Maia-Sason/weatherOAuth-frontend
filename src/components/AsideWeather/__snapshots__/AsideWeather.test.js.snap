// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AsideWeather/> matches snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-container-1 MuiPaper-elevation10 MuiPaper-rounded"
  >
    <div
      class="MuiBox-root MuiBox-root-8 makeStyles-headerBox-2"
    >
      <p
        class="MuiTypography-root makeStyles-header-3 MuiTypography-body1"
      >
        Feels Like:
      </p>
      <p
        class="MuiTypography-root makeStyles-header-3 MuiTypography-body1"
      >
        Â°
      </p>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiBox-root MuiBox-root-9 makeStyles-precipitation-6"
    >
      <svg
        class="makeStyles-rainDrop-5"
      >
        drop-percentage.svg
      </svg>
      <p
        class="MuiTypography-root makeStyles-headerB-4 MuiTypography-body1"
      >
        NaN
        %
      </p>
    </div>
    <hr
      class="MuiDivider-root"
    />
    <div
      class="MuiBox-root MuiBox-root-10 makeStyles-wind-7"
    >
      <svg
        class="makeStyles-rainDrop-5"
      >
        wind-2.svg
      </svg>
      <p
        class="MuiTypography-root makeStyles-headerB-4 MuiTypography-body1"
      >
        mph
      </p>
    </div>
  </div>
</div>
`;
